### ISSUE TITLE
StructureExtractor Module

### ISSUE THREAD

**Issue by callmest** [BorgwardtLab/PST#1]
Thanks for your wonderful job! My downstream tasks make a good improvement based on this work!
I read your PST model source code and wonder why there needs to be a **BN** rather than the **LN** layer.
![1711090971352](https://github.com/BorgwardtLab/PST/assets/66674446/0fe97e55-124f-4cf0-a079-929eb83eaa96)

**Comment by claying** [comment]
Thanks a lot for your interest in our work, @callmest ! We used batch norms in our structure extractor based on the original code of SAT (https://github.com/BorgwardtLab/SAT/blob/main/sat/layers.py), as for graph neural networks, batch norms seem to work better than LN. But for protein structures, we did not do any ablation study of this, so feel free to test it.

**Comment by callmest** [comment]
Thanks! It seems that **gnn.radius_graph** was used to generate edge_index, do you have any extra study on how to generate edge_index in this work? Or is this a common way to get protein structure from this method?

**Comment by claying** [comment]
In section 4.4.1 of our paper (https://arxiv.org/pdf/2401.14819.pdf), we have tried more complex edge attributes such as distance information. While this does improve pretraining performance, the performance of the model on downstream tasks decreases. That's why we only use such a simple epsilon-neighborhood graph requiring minimal structural information. Note that this construction strategy should also be robust to structures obtained by different methods, such as AlphaFold and ESMFold.

**Comment by callmest** [comment]
Thanks! Indeed, the PST model performed well on my downstream task! I should read the paper more carefully.
